config/routes.rb

```diff
@@ -4,7 +4,7 @@
     get '/users'                   => 'users#index'
     get '/users/:id'               => 'users#show'
     get '/users/:user_id/posts'     => 'posts#index'
-    get '/users/:user_id/posts/:id' => 'posts#show'
+    get '/users/:user_id/posts/:post_id' => 'posts#show'
 end
```

Comment:
- [OPTIONAL] Great job so far @Atifzada Your project works great. There are just a few changes I will suggest you make :wink: 
- [ ] The users and posts endpoints appear to be correct and maintain a RESTful URL structure. However, it is important to note that the appropriate HTTP verbs should be used to match the corresponding actions in a RESTful design. Here's an updated version of the routes using the conventional HTTP verbs:

```ruby
resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :show]
end
```
- The code above uses the `resources method`, which generates the conventional RESTful routes for the users and posts resources. The `only` option is used to specify the allowed actions for each resource. :+1:

Reply:
Thank you for the suggestions 
sure I will work on it